pkg/agent/flannel/setup.go

```diff
@@ -146,12 +146,17 @@ func waitForPodCIDR(ctx context.Context, nodeName string, nodes typedcorev1.Node
 	return nil
 }
 
-func createCNIConf(dir string) error {
+func createCNIConf(dir string, nodeConfig *config.Node) error {
 	logrus.Debugf("Creating the CNI conf in directory %s", dir)
 	if dir == "" {
 		return nil
 	}
 	p := filepath.Join(dir, "10-flannel.conflist")
+
+	if nodeConfig.AgentConfig.FlannelCniConfFile != "" {
+		logrus.Debugf("Using %s as the flannel CNI conf", nodeConfig.AgentConfig.FlannelCniConfFile)
+		return util.CopyFile(nodeConfig.AgentConfig.FlannelCniConfFile, p)
```

Comment:
is the fact that we're copying the file into place instead of using it as-is going to surprise anyone? Is there a downside to using the file in its specified location?

Reply:
The CNI binary expects the config to be in that specific path. We can change that but it feels more complicated. Let me have a look at it

Reply:
Given that, copying it makes sense. Thanks!